jwst/datamodels/model_base.py

```diff
@@ -237,16 +238,16 @@ def __init__(self, init=None, schema=None,
                 if 'datatype' in subschema:
                     setattr(self, attr, value)
 
-
-    def __repr__(self):
-        import re
-
-        buf = ['<']
+    @property
+    def model_type(self):
         match = re.search(r"(\w+)'", str(type(self)))
         if match:
-            buf.append(match.group(1))
-        else:
-            buf.append("DataModel")
+            return match.group(1)
+        return "DataModel"
```

Comment:
A better way to do this without `re`:
```
    @property
    def model_type(self):
        return self.__class__.__name__
```

Reply:
Or just despense with the property altogether and use `self.__class__.__name__` elsewhere.

Reply:
I would rather use a property, then spreading dunder-usage throughout code. And this encapsulates the definition.

About using re/type vs. the dunder, this is just a personal preference of trying to avoid dunders vs. using available public api. For this particular case, using dunder attributes is fine.

Reply:
Offline discussion: Property is fine, but make is private, `_model_type`. There is an issue of being able to pretend to set this, but that is a larger issue with the `DataModel` api.